### PR TITLE
#1355 Add a setting to hide the usernames from the dropdown list

### PR DESCRIPTION
**Pull request by DimitrisGk-iel**
**What's in the PR**
webanno#1355 - Add a setting to hide the usernames from the dropdown list.
Reads from the settings.properties file the configuration settings userSelection.hideUsers, userSelection.minLengthCharacters and modifies the behavior of the dropdown list accordingly.
Updates the user guide User section.
Introduces the new settings to the admin guide General Settings table.

**How to test manually**
Add the userSelection.hideUsers=true setting into the settings.properties file.
Go to the Users tab and click into the textField to add a user.  You shouldn't see any dropdown list with users.
Write the username of the user you wish to add. Only this username should be visible in the dropdown list.

**Documentation**
* [ ] PR updates documentation

### CODE REVIEW

**Comment by DimitrisGk-iel** [comment]
I am sorry i completely missed the section in WebAnno site where the process of creating a PR is described. As you mentioned the previous PR had issues that were hard to fix. Thus, i closed it and created this new PR.

File: webanno-doc/src/main/resources/META-INF/asciidoc/admin-guide/settings_general.adoc [hunk 1]
```diff
@@ -43,5 +43,15 @@
 | Whether regular users can access their own profile to change their password and other profile information. This setting has no effect if WebAnno is running in pre-authentication mode.
 | false
 | true
+
+| userSelection.hideUsers
+| Whether the system's usernames will be shown into the dropdown list at the Users tab. In order for the Project manager to add a user into an annotation project she has to write into the same text Field the annotator’s full username. When there is a full match this username (and only this) is displayed in the dropdown list where the Project Manager can select it and add it to her project.
+| false
+| true
+
+| userSelection.minLengthCharacters
```
Comment (reckart) [inline webanno-doc/src/main/resources/META-INF/asciidoc/admin-guide/settings_general.adoc]: I don't quite understand what this setting is for. I though the idea was that if `hideUsers` is true, the manager has to enter the entire name of the user to add into the field. But why is the min-length then needed. The dropdown should only ever show either nothing or the username entered by the user if there is a user with that name?

**Comment by reckart** [comment]
Ok, now we do proper server-side filtering :)

**Comment by reckart** [comment]
@DimitrisGk-iel I'm going to merge this once Jenkins gives the thumbs up.

**Comment by reckart** [comment]
Jenkins, can you test this please?

**Pull request merged**